Makefile

```diff
@@ -42,48 +50,62 @@ patch-dev: superset-clean
 	ln -s "$(PWD)/$(PATCH_SOURCE_DIR)/superset/superset_config_dev.py" "$(SUPERSET_DIR)/superset_config.py"; \
 
 # Create docker image
-.PHONY: superset-build
-superset-build: patch
-	docker build -t $(SUPERSET_IMAGE_NAME):$(VERSION) -f superset/contrib/docker/Dockerfile superset
+.PHONY: patch
+build: patch
+	docker build -t $(DOCKER_IMAGE_NAME):$(VERSION) -f superset/contrib/docker/Dockerfile $(SUPERSET_DIR)
+
+.PHONY: docker-validate
+docker-validate:
+	@if [ -z "$(DOCKER_USERNAME)" ]; then \
+		echo "DOCKER_USERNAME variable cannot be empty."; \
+		exit 1; \
+	fi; \
+	if [ -z "$(DOCKER_PASSWORD)" ]; then \
+		echo "DOCKER_PASSWORD variable cannot be empty."; \
+		exit 1; \
+	fi
 
-# Push the superset docker image, based on .ci/Makefile.main
-superset-docker-push: docker-login superset-build
+.PHONY: docker-login
+docker-login: docker-validate
```

Comment:
Isn't this a copy-paste from [src-d/ci/Makefile::docker-login](https://github.com/src-d/ci/blob/v1/Makefile.main#L180) that could be removed if using it?